cmd/query/app/token_propagation_handler.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2019 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
+
+import (
+	"net/http"
+	"strings"
+
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/storage/spanstore"
+)
+
+func bearTokenPropagationHandler(logger *zap.Logger, h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		authHeaderValue := r.Header.Get("Authorization")
+		if authHeaderValue != "" {
+			headerValue := strings.Split(authHeaderValue, " ")
+			token := ""
+			if len(headerValue) == 2 {
+				token = headerValue[1]
```

Comment:
The way it is, you are probably propagating non-bearer authorization as well, such as `Basic`. If this is what you want, you should rename the functions from `bearerTokenPropagation` to `authorizationHeaderPropagation`. 

If you didn't intend to propagate the basic auth as well, check explicitly for the authorization type (`"Bearer" == headerValue[0]`).